### ISSUE TITLE
VCV Rack using tons of resources - Loud Fan, CPU Load, Extreme Battery Depletion

### ISSUE THREAD

**Issue by JWhiteUX** [VCVRack/Rack#992]
I'm currently using VCV Rack on a MacBook Pro.

OSX 10.13.3 (17D102)
2.2 GHz Intel Core i7
16 GB 1600 MHz DDR3
Intel Iris Pro 1536 MB

I'm experiencing the following issues.

1. Excessive fan noise and heat
2. CPU load spikes
3. Clipping/stuttering when switching between applications or windows
4. Extreme(!) battery depletion (at 100% I have about 40-50 minutes of Rack time before I hit ~10%)

For what it's worth, I don't have this issue with NI Reaktor 6 or Ableton, granted they're not opensource or community built, so that explains some stuff right off the bat, but the resource allocation of VCVRack needs some serious optimizations.

**Comment by JWhiteUX** [comment]
"research beyond basic performance measurements"

So… Does that mean these screens are of little or no help?

<img width="292" alt="screen shot 2018-06-07 at 3 52 02 pm" src="https://user-images.githubusercontent.com/12961108/41132533-963b35c4-6a76-11e8-94d2-db00092a44cf.png">
![screen shot 2018-06-07 at 4 16 26 pm](https://user-images.githubusercontent.com/12961108/41132536-9a686432-6a76-11e8-8c21-97bad83f8214.png)

![screen shot 2018-06-07 at 4 16 35 pm](https://user-images.githubusercontent.com/12961108/41132547-a459b5f4-6a76-11e8-8207-60964d70bba5.png)

**Comment by AndrewBelt** [comment]
Yes, believe me, this is not new information.

**Comment by JWhiteUX** [comment]
Unfortunately, I can't. I disconnected all wires and removed a couple modules before sharing the screenshots.

**Comment by JWhiteUX** [comment]
@dirkleas Heres a file that with only 6 patches, is destroying my macbook (16GB)
[resource-hob-6cables.vcv.zip](https://github.com/VCVRack/Rack/files/2082967/resource-hob-6cables.vcv.zip)

**Comment by dirkleas** [comment]
@JWhiteUX, I loaded the patch on my macbook pro 16G late '16 w/ latest rack built from source which has the performance meter -- patch ran as expected at about 50% on the meter (also randomized the sequencer a couple of times to add some sonic variation). it would be interesting to see what you get if you can replicate latest build from source w/ meter

**Comment by JWhiteUX** [comment]
I'll look into installing the latest build. That VCV file is admittedly about 1/2 of what I was originally doing. I had about 4 Branches, StepMatrix, 3oscs, 2lfos, a couple filters, and whatever I left out that you see in the file I shared. I had a rings braids and some terrible knockoff MI module that has very low volume out. Not sure why, but VCV rack requires that I crank the volume up. I don't have this issue with any of my other software though.

The volume issue might be a completely different story

**Comment by JWhiteUX** [comment]
@dirkleas — Where are you seeing the latest build? I only see 0.6.0 in releases. Are you compiling locally?

**Comment by dirkleas** [comment]
@JWhiteUX, to get the pre-release metering capability you'll need to compile Rack from source to get latest commits

**Comment by gabrielaizcorbe** [comment]
@AndrewBelt I was browsing the community, and I was thinking, do you recommend any kind of CPU for VCV, I mean, when you developed this app ... you did it thinking in a certain architecture? is the same AMD or Intel for the app? Thanks for the app, your time and your answers!

**Comment by AndrewBelt** [comment]
@gabrielaizcorbe No, I designed it for general x86_64 + SSE2 processors.

**Comment by maddanio** [comment]
I wanted to comment on the merged issue, but I am not allowed (could I become a colaborator?).
Anyway, fwiw, I did some profiling of vcvrack v1, and I found that, besides the gui code taking a third of the time, another third is taken by waiting on spin locks. since vcvrack is using two worker threads in that instance, each of which take a third of the cpu (the other as said above is gui) this means effectively one thread would be just as effective.
maybe if I have a good idea/better understanding I will try improving the threading code.

**Comment by modman2019** [comment]
Yes it would be nice if there was a way to reduce the resources this uses especially in the upcoming vst version.  I dont have a dedicated graphics card so I think that is causing an issue.  Maybe if we can use Asio drivers that will help? So far I can use Wasapi.
